ledger-core/virtual/integration/outgoing_test.go

```diff
@@ -0,0 +1,221 @@
+	authService.IsMessageFromVirtualLegitimateMock.Set(func(_ context.Context, payloadObj interface{}, sender reference.Global, pr pulse.Range) (mustReject bool, err error) {
+		return false, nil
+	})
+
+	server.ReplaceAuthenticationService(authService)
+
+	server.Init(ctx)
+	server.IncrementPulseAndWaitIdle(ctx)
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+	var (
+		barIsolation = contract.MethodIsolation{
+			Interference: contract.CallTolerable,
+			State:        contract.CallDirty,
+		}
+		fooIsolation = contract.MethodIsolation{
+			Interference: contract.CallTolerable,
+			State:        contract.CallDirty,
+		}
+
+		classB        = gen.UniqueGlobalRef()
+		objectAGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+
+		objectBGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+
+		callOutgoing = server.RandomLocalWithPulse()
+
+		outgoingCallRef = reference.NewRecordOf(objectAGlobal, callOutgoing)
+	)
+
+	Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+
+	// add ExecutionMocks to runnerMock
+	{
+		outgoingCall := execution.NewRPCBuilder(outgoingCallRef, objectAGlobal).
+			CallMethod(objectBGlobal, classB, "Bar", []byte("123")).
+			SetInterference(barIsolation.Interference).
+			SetIsolation(barIsolation.State)
+		objectBExecutionMock := runnerMock.AddExecutionMock("Foo")
+		objectBExecutionMock.AddStart(
+			func(ctx execution.Context) {
+				logger.Debug("ExecutionStart [A.Foo]")
+				assert.Equal(t, objectAGlobal, ctx.Object)
+				assert.Equal(t, callOutgoing, ctx.Request.CallOutgoing)
+			},
+			&execution.Update{
+				Type:     execution.OutgoingCall,
+				Error:    nil,
+				Outgoing: outgoingCall,
+			},
+		)
+		objectBExecutionMock.AddContinue(
+			func(result []byte) {
+				logger.Debug("ExecutionContinue [A.Foo]")
+				require.Equal(t, []byte("finish B.Bar"), result)
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("finish A.Foo"), objectAGlobal),
+			},
+		)
+
+		runnerMock.AddExecutionClassify("Foo", fooIsolation, nil)
+	}
+
+	callCount := 0
+
+	firstPulse := server.GetPulse().PulseNumber
+
+	expectedToken := payload.CallDelegationToken{
+		TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+		PulseNumber:       server.GetPulse().PulseNumber,
+		Callee:            objectAGlobal,
+		Outgoing:          outgoingCallRef,
+		DelegateTo:        server.JetCoordinatorMock.Me(),
+	}
+
+	{
+		typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+			assert.Equal(t, firstPulse, report.OrderedPendingEarliestPulse)
+			assert.Equal(t, int32(1), report.OrderedPendingCount)
+			return false
+		})
+		typedChecker.VDelegatedCallRequest.Set(func(request *payload.VDelegatedCallRequest) bool {
+			assert.Zero(t, request.DelegationSpec)
+			assert.Equal(t, objectAGlobal, request.Callee)
+			assert.Equal(t, payload.BuildCallFlags(fooIsolation.Interference, fooIsolation.State), request.CallFlags)
```

Comment:
why do you check flags? I do think this test isn't about isolation.

Reply:
removed extra checks.